projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts

```diff
@@ -108,4 +110,43 @@ export class NavigationUIComponent implements OnDestroy {
       this.subscription.unsubscribe();
     }
   }
+
+  private alignWrapperToRightIfStickOut(node: HTMLElement) {
+    const wrapper = <HTMLElement>node.querySelector('.wrapper');
+    const navBar = <HTMLElement>this.elemRef.nativeElement;
+    if (wrapper) {
+      this.renderer.removeStyle(wrapper, 'margin-left');
+      if (
+        wrapper.offsetLeft + wrapper.offsetWidth >
+        navBar.offsetLeft + navBar.offsetWidth
+      ) {
+        this.renderer.setStyle(
+          wrapper,
+          'margin-left',
+          `${node.offsetWidth - wrapper.offsetWidth}px`
+        );
+      }
+    }
+  }
+
+  private validateWrappersAndAlignToRightIfStickOut() {
```

Comment:
maybe we should think about shortening this name - `validateWrappersAndAlignToRightIfStickOut()`

Reply:
changed to alignWrappersToRightIfStickOut